packages/flame/lib/src/components/component.dart

```diff
@@ -493,22 +517,26 @@ class Component {
     components.forEach(remove);
   }
 
-  Future<void>? _load() {
+  Future<void>? _startLoading() {
+    assert(_state == _initial);
     assert(_parent != null);
-    assert(_state == LifecycleState.uninitialized);
-    _state = LifecycleState.loading;
+    assert(_parent!.findGame() != null);
+    assert(_parent!.findGame()!.hasLayout);
```

Comment:
We are starting to have a quite some assertions here, would it be good to have a message to give more context about them?

Reply:
I'm trying to distinguish between assertions that trigger when the user does something wrong -- these assertions will have proper messages, -- versus the assertions that can only trigger if we did something wrong, i.e. if there is a bug in our code. The latter kind of assertions are not possible to trigger, at least we don't know how they can be possibly triggered (if we did, we'd be able to fix that). So their function is to (1) state the pre-/post-conditions of a method (for example, at a start of this method we believe that the parent is set, and that the game is laid out), and (2) guard against possible problems during future refactors.